diff --git a/mwm.ahk b/mwm.ahk
index 7ab1ec7..b376fcd 100644
--- a/mwm.ahk
+++ b/mwm.ahk
@@ -7,6 +7,7 @@ KeyHistory(0), ListLines(false), ProcessSetPriority("H")
 
 #include *i lib\miguru\miguru.ahk
 #include *i lib\Popup.ahk
+#include macros.ahk
 
 GroupAdd("MIGURU_AUTOFLOAT", "Microsoft Teams-Benachrichtigung" " ahk_exe Teams.exe"                                                  )
 GroupAdd("MIGURU_AUTOFLOAT", "Microsoft Teams-Notification"     " ahk_exe Teams.exe"                                                  )
@@ -22,9 +23,13 @@ GroupAdd("MIGURU_AUTOFLOAT", "WinMerge"
 GroupAdd("MIGURU_DECOLESS",                                     " ahk_exe qutebrowser.exe"                                            )
 GroupAdd("MIGURU_DECOLESS",                                     " ahk_exe alacritty.exe"                                              )
 GroupAdd("MIGURU_DECOLESS",                                     " ahk_exe msrdc.exe"                " ahk_class RAIL_WINDOW"          )
+GroupAdd("MIGURU_DECOLESS",                                     " ahk_exe mintty.exe"                                                 )
 
 GroupAdd("MIGURU_IGNORE",                                       " ahk_exe msrdc.exe"                                                  )
 GroupAdd("MIGURU_IGNORE",    "WinUI Desktop"                    " ahk_exe PowerToys.Peek.UI.exe"    " ahk_class WinUIDesktopWin32WindowClass")
+GroupAdd("MIGURU_IGNORE",                                       " ahk_exe mstsc.exe"                                                  )
+GroupAdd("MIGURU_IGNORE",                                       " ahk_exe VirtualBox.exe"                                             )
+GroupAdd("MIGURU_IGNORE",                                       " ahk_exe SnippingTool.exe"                                           )
 
 if !IsSet(MiguruWM) {
     prog := RegExReplace(A_ScriptName, "i)\.ahk$", ".exe")
@@ -36,14 +41,8 @@ if !IsSet(MiguruWM) {
 
 layouts := [
     TallLayout(),
-    WideLayout(),
     FullscreenLayout(),
     FloatingLayout(),
-    ColumnLayout(),
-    RowLayout(),
-    ThreeColumnLayout(),
-    TwoPaneLayout(),
-    SpiralLayout(),
 ]
 
 mwm := { __Call: (name, params*) => } ; Ignore requests while mwm isn't ready yet
@@ -70,7 +69,7 @@ mwm := MiguruWM({
     followWindowToWorkspace: false,
     followWindowToMonitor: false,
 
-    focusWorkspaceByWindow: true,
+    focusWorkspaceByWindow: false,
 
     delays: {
         retryManage: 100,
@@ -124,8 +123,8 @@ mod1 := "Alt"
 *Enter::mwm.Do("swap-window", { with: "master-or-mru" })
 *Space::mwm.Do("cycle-layout", { value: layouts })
 
-*vk01::MoveActiveWindow()
-*vk02::ResizeActiveWindow()
+^*vk01::MoveActiveWindow()
+^*vk02::ResizeActiveWindow()
 
 #Hotif GetKeyState(mod1, "P") and GetKeyState("Shift", "P")
 
